bee-message/src/parents.rs

```diff
@@ -51,19 +144,106 @@ impl Packable for Parents {
     }
 
     fn unpack<R: Read + ?Sized>(reader: &mut R) -> Result<Self, Self::Error> {
-        let parents_len = u8::unpack(reader)? as usize;
+        let _parents_len = u8::unpack(reader)? as usize;
+
+        // const MESSAGE_PARENTS_RANGE: RangeInclusive<usize> = 1..=8;
+
+        // if !MESSAGE_PARENTS_RANGE.contains(&parents_len) {
+        //     return Err(Error::InvalidParentsCount(parents_len));
+        // }
+
+        // NB: The code doesn't depend on `parents_len`, or its correctness anymore. Instead, the
+        // code errors out, if it can not unpack at least 1, or even more than 8 parents. Each
+        // `parents` variable is a distinct type according to the builder stage we're in.
```

Comment:
You are relying on external factors (remember you can have many things packed one after the other, there could be lots of data in the reader after your message) where the deserialisation of a type should be self-contained.

Reply:
yes, you're right here as well. And that's something I wondered about actually during implementation. Good point!